pkg/container/runtime/cri.go

```diff
@@ -105,10 +106,12 @@ func getRuntimeClient(addr string) (pb.RuntimeServiceClient, *grpc.ClientConn, e
 }
 
 func getRuntimeClientConnection(addr string) (*grpc.ClientConn, error) {
-	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(5*time.Second))
```

Comment:
Did you consider to add proper context propagation here, i.e. adding a context to, say, `StopContainer` and friends to use it here?